craft_parts/packages/deb.py

```diff
@@ -481,10 +481,8 @@ def refresh_stage_packages_list(cls, *, cache_dir: Path, target_arch: str):
         stage_cache_dir, _ = get_cache_dirs(cache_dir)
         stage_cache_dir.mkdir(parents=True, exist_ok=True)
 
-        with AptCache(
-            stage_cache=stage_cache_dir, stage_cache_arch=target_arch
-        ) as apt_cache:
-            apt_cache.update()
+        with AptCache(stage_cache=stage_cache_dir, stage_cache_arch=target_arch) as _:
+            pass
```

Comment:
Agreed, just took a quick look at the implementation and not sure why __init__ is somewhat different to __enter__. This could be extracted into a method called after or could be implemented using a classmethod if the regression risk is high to just match as close as possible.

Reply:
I agree it would be best to expose it as its own interface. However cache doesn't really need to be initialized until use and this could be dropped entirely.  I originally did that, but it impacted a number of tests expecting refresh to cause the cache to exist, even if unused.  To limit the scope of changes in this PR I compromised with this step. But I can update it however you wish :D

Reply:
Maybe we could extract the **enter** contents to a new method and call this method from **enter** to preserve existing functionality, and call the method directly instead of using the empty context.

Reply:
Resolved by removing the stage package list updating interface. This can be re-added later if needed by a different package management system.